packages/apm/src/hubextensions.ts

```diff
@@ -19,6 +19,19 @@ function traceHeaders(this: Hub): { [key: string]: string } {
   return {};
 }
 
+/**
+ * {@see Hub.getTransaction}
+ */
+function getTransaction(this: Hub, callback: (transaction: Transaction) => void): void {
+  const scope = this.getScope();
+  if (scope) {
+    const span = scope.getSpan() as Transaction;
```

Comment:
So this isnt actually correct - this just gets the current span, no?

I need the current _transaction_, aka the first parent span that is marked as a transaction, because I need to be able to mutate its metadata based on the framework or other information not available when transaction is created.

@untitaker can give more context as I need this both in my Gatsby work as well as in our GraphQL Python code.

Reply:
@dcramer I think the browser SDK may work slightly different than Python as in, it doesn't always put the current span onto the scope. So this *may* be correct. But again, I am not sure.

@HazAT let's talk on Friday abt this

Reply:
@untitaker Yeah in theory you can set any span on the Scope (not just transactions) right now in JS we don't have a case where we set anything besides the Transaction on the Scope. I remember our conversation where it makes sense to set a child Span on the Scope.

So, we could change that function to always return just the Transaction and not more and for cases that we know we need to Span (in Python) we rely on the old method of just fetching the Span from the Scope.

We could add a convenient function on the Span to get the Transaction (`span.getTransaction()`) in cases where we need both. We can use this function also then in `Sentry.getTransaction` to always make sure we return the Transaction.

Reply:
having it on the span seems valuable (though not sure if more or less correct than just having it on the scope itself)

Reply:
I changed it so it's called `getSpan` and a `Span` has a method called `getTransaction` to make it more clear.
Since we would consider this already a more advanced use-case, I guess it's fair for the user to understand the relationship between a Span and a Transaction